src/ComparableDiscreteSystem.ts

```diff
@@ -0,0 +1,55 @@
+import ndarray from "ndarray";
+import { DiscreteSystem } from "./DiscreteSystem.ts";
+import { ComparableSystem } from "./interfaces/ComparableSystem.ts";
+import { particleStateToArray } from "./utils.ts";
+import { Random } from "@reside-ic/random";
+import { ComparableDiscreteGenerator } from "./interfaces/ComparableDiscreteGenerator.ts";
+
+export class ComparableDiscreteSystem<TShared, TInternal, TData>
+    extends DiscreteSystem<TShared, TInternal>
+    implements ComparableSystem<TData>
```

Comment:
We can implement multiple interfaces, right? If so this feels exactly what we need

Reply:
yep!